src/types.ts

```diff
@@ -637,17 +637,15 @@ export class Signature {
   }
 
   /** Converts this signature to a string. */
-  toString(includeThis: bool = false): string {
+  toString(): string {
     var sb = new Array<string>();
     sb.push("(");
     var index = 0;
     var thisType = this.thisType;
     if (thisType) {
-      if (includeThis) {
-        sb.push("this: ");
-        sb.push(thisType.toString());
-        index = 1;
-      }
+      sb.push("this: ");
+      sb.push(thisType.toString());
```

Comment:
It seems this cause to recursively calls: Signature#toString -> Type#toString() -> Signature#toString ...
May be should revert back `includeThis: bool = false`?

Reply:
Are you sure about this? `thisType` is never a signature I think.

Reply:
added an assertion to be sure :)